### PR TITLE
[chassis] Enhance test_default_route.py for chassis

### PR DESCRIPTION
**Pull request by abdosi**
What I did :

Enhance `test_default_route.py` for chassis.

How I did:

- Concept of selecting upstream duthost and verify test case on that
- Depends on this PR https://github.com/sonic-net/sonic-mgmt/pull/6945 for common utility function.
- Some of Core Router even though T3 we might not expect Default Route based on Device Type.
- For verifying Default route for downstream duthost in chassis via iBGP we will need to enhance test case further.  Will create test gap for that.

How I verify:
Verified on both T1 and chassis topology.

### CODE REVIEW

**Comment by abdosi** [comment]
cc @judyjoseph

**Review by wenyiz2021 — APPROVED** [review]

File: tests/route/test_default_route.py [hunk 1]
```diff
@@ -112,13 +115,13 @@ def verify_default_route_in_app_db(duthost, tbinfo, af, uplink_ns):
...
 
-def test_default_route_set_src(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_asic_index, tbinfo):
+def test_default_route_set_src(duthosts, tbinfo):
     """
     check if ipv4 and ipv6 default src address match Loopback0 address
 
     """
-    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
-    asichost = duthost.asic_instance(enum_asic_index)
+    duthost = find_duthost_on_role(duthosts, get_upstream_neigh_type(tbinfo['topo']['name']) , tbinfo)
```
Comment (wenyiz2021) [inline tests/route/test_default_route.py:123]: where is `find_duthost_on_role`?

Comment (abdosi) [reply tests/route/test_default_route.py:123]: it has been merged as part of everflow PR

**Pull request merged**